chirp/drivers/alinco_dr735t.py

```diff
@@ -0,0 +1,368 @@
+
+        command = b"AL~RESET\r\n"
+        self.pipe.write(command)
+        self.pipe.read(len(command))  # command + OK\r\n
+        self.pipe.read(4)
+
+    def sync_in(self):
+        try:
+            self._mmap = self.do_download()
+        except Exception as exc:
+            raise errors.RadioError(f"Failed to download from radio: {exc}")
+        self.process_mmap()
+
+    def sync_out(self):
+        try:
+            self.do_upload()
+        except Exception as exc:
+            raise errors.RadioError(f"Failed to download from radio: {exc}")
+
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def get_memory(self, number):
+        _mem = self._memobj.memory[number]
+        mem = chirp_common.Memory()
+        mem.number = number                 # Set the memory number
+        mem.freq = int(_mem.frequency)
+        mem.name = "".join([CHARSET[_mem.name[i]] for i in range(6)]).strip()
+
+        mem.tmode = self.TONE_MODE_MAP[int(_mem.subtone_selection)]
+        mem.duplex = self.SHIFT_DIR_MAP[_mem.shift_direction]
+        mem.offset = _mem.shift
+
+        mem.rtone = ALINCO_TONES[_mem.rx_tone_index]
+        mem.ctone = ALINCO_TONES[_mem.tx_tone_index]
+        mem.dtcs = chirp_common.DTCS_CODES[_mem.dcs_index]
+        mem.power = self.POWER_MAP[_mem.power_index]
+        mem.skip = 'S' if bool(_mem.skip) else ''
+        mem.mode = self.MODE_MAP[int(_mem.mode)]
+
+        self._get_extra(_mem, mem)
+
+        if _mem.used == 0:
+            mem.empty = True
+        return mem
+
+    def set_memory(self, mem):
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.memory[mem.number]
+
+        def find_key_in(d: dict, target_val):
+            for k, v in d.items():
+                if v == target_val:
+                    return k
+
+        if not mem.empty:
+            mapped_name = [CHARSET.index(' ').to_bytes(1, 'little')]*6
+            for (i, c) in enumerate(mem.name.ljust(6)[:6].upper().strip()):
+                if c not in chirp_common.CHARSET_UPPER_NUMERIC:
+                    c = " "  # just make it a space
+                mapped_name[i] = CHARSET.index(c).to_bytes(1, 'little')
+            _mem.frequency = int(mem.freq)
+            _mem.name = b''.join(mapped_name)
+            _mem.mode = find_key_in(self.MODE_MAP, mem.mode)
+            _mem.subtone_selection = find_key_in(self.TONE_MODE_MAP, mem.tmode)
+            _mem.shift = mem.offset
+            _mem.used = 0x00 if mem.empty else 0x55
+            _mem.power_index = self.POWER_MAP.index(
+                mem.power) if mem.power in self.POWER_MAP else 0
+            _mem.skip = 0x01 if mem.skip == "S" else 0x00
+            try:
+                _mem.rx_tone_index = ALINCO_TONES.index(
+                    mem.rtone if mem.rtone else 67.0)
```

Comment:
Why are you doing `if mem.rtone else`? There should be no case where `mem.rtone` is falsey.

Reply:
Hi Dan, thanks for looking at my PR.  One of the driver tests was failing on mem.rtone being None in some cases it seemed, although I couldn't really tell given the test output was a little hard to understand since I'm new to the project. I can remove the falsesy check since the exception should catch any wrong value (I added the try later). Happy to change that.

Reply:
Okay, yeah, I'm not sure when/where that would be but I'd rather not have the implication here and handle any failure to find the tone values with the exception handler. Thanks!